src/circuit_proof/r1cs.rs

```diff
@@ -57,6 +58,7 @@ impl LinearCombination {
 }
 
 pub struct ConstraintSystem {
+    transcript: Transcript,
```

Comment:
This is taking ownership of the `Transcript` object directly, but I think we want to take ownership just of a mutable borrow of a `Transcript` owned by some higher-level function.  So it seems like the definition should be

```rust
pub struct ConstraintSystem<'a> {
    transcript: &'a mut Transcript,
    /* ... */
}
```